### ISSUE TITLE
Segfault with sophia engine

### ISSUE THREAD

**Issue by lekster** [tarantool/tarantool#1508]
Tarantool 1.6.8-692-g584c3ae
Ubuntu 14.04.4
php-tarantool release 0.1.0

init.lua

```
local console = require('console')
console.listen('127.0.0.1:3312')
box.cfg {
    listen = 3301,
    logger = 'tarantool.log',
}

s =  box.space.test
if not s then
    s = box.schema.space.create('test', { engine = 'sophia' })
    p = s:create_index('primary', {type = 'tree', parts = {1, 'NUM'}})
end

box.schema.user.grant('guest', 'read,write,execute', 'universe', nil, {if_not_exists=true})
```

php client

```
$tnt = new Tarantool('vps172202180', 3301);
try {
	$t = $tnt->upsert('test', array(99999, '1'), array(
		array(
			"field" => 1,
			"op" => "+",
			"arg" => 1
			)
		)
	);
	$t = $tnt->select("test");
	var_dump($t);
} catch (Exception $e) {
	echo "Exception: ", $e->getMessage(), "\n";
}

```

first run php script - OK, record inserted in tarantool
second - server crash in segfault

memtx works without problem.

**Comment by rtsisyk** [comment]
Duplicate of #1077 which is already fixed in 1.7.x.

Your example produces the following error in the latest 1.7.x:

```
2016-05-31 22:55:50.793 [15315] coeio tuple_update.cc:946 E> UPSERT operation failed:
2016-05-31 22:55:50.793 [15315] coeio tuple_update.cc:291 E> ER_ARG_TYPE: Argument type in operation '+' on field 1 does not match field type: expected a NUMBER
```

**Comment by rtsisyk** [comment]
I have to admit that Sophia is not production-ready in 1.6.x. I don't think that back-porting this fix to 1.6.x makes sense. Sorry.

Currently our team works hardly to resolve all issues with disk engine in 1.7.x branch.
Please be patient, we will manage to release 1.7 alpha soon.
